testgrid/cmd/gatherer/README.md

```diff
@@ -0,0 +1,18 @@
+#GitHub Issue Gatherer
+The GitHub Issue Gatherer collects issues and identifies issues that reference TestGrid tests. This is so that TestGrid can hyperlink failing or flaky tests to open issues.
+
+It has three parts: an Issue Scraper, a Target Identifier, and a Proto Writer.
+
+## Issue Scraper (issuescraper.go)
+The Issue Scraper uses the GitHub v3 API to collect every issue in a repository, and every comment left on those issues.
+
+GitHub calls Pull Requests "Issues", but this scraper doesn't.
+
+This scraper uses a caching layer, [ghCache](https://github.com/kubernetes/issues.json/tree/master/ghproxy/ghcache), to minimize repetitive API calls.
+
+The scraper will automatically [depaginate](https://developer.github.com/v3/guides/traversing-with-pagination/) results and store them in a slice of GitHubIssues. That type definition can be found in the code.
```

Comment:
Is this unique to your implementation?

Reply:
Depagination isn't, but I think that caching unchanged pages might be.

From what I can tell,
- prow/github utility uses the v4 API, not the RESTful API.
- prow/ghCache caches unchanged pages, but doesn't work with the v4 API

So it seemed like I could either rewrite the API calling in v3, or rewrite the caching in v4. I decided to rewrite the calling.

Have we used ghCache with an existing utility somewhere else? I'd love to see an example, if so.

Reply:
The GitHub client in Prow uses V3 as well as V4